src/App.tsx

```diff
@@ -1,39 +1,68 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
 
+import { Clock } from './clock';
+
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends Component {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  threeSecondTimer = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount() {
+    this.threeSecondTimer = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+      this.setState({ hasClock: false });
+    });
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    document.addEventListener('click', () => {
+      this.setState({ hasClock: true });
+    });
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentDidUpdate(_prevProps: State, prevState: State) {
+    // eslint-disable-next-line
+    this.state.hasClock && console.debug(
+      `Renamed from ${prevState.clockName} to ${this.state.clockName}`,
+    );
+  }
```

Comment:
Something went wrong 🥲
Try hide and show your clock a few times and check the console 
<img width="368" alt="Screenshot 2022-09-15 at 17 53 13" src="https://user-images.githubusercontent.com/47606379/190436433-0df50a53-5f1a-4368-8809-f44690f2caa3.png">

(try use componentDidUpdate inside Clock)